Add `environment` command

vernig opened the pull request:
Add `infra:environment` commands: 

* `infra:environment:set`: set the environment (i.e. Pulumi stack)
* `infra:environment:new`: create a new environment (i.e. Pulumi stack) for the projected selected in the Twilio CLI
* `infra:environment:get`: get the current environment (i.e. Pulumi stack)

Commit:
Add deployment log helper lib

This lib is used to interact with the local `.twilio-infa` file

Commit:
Add get current deployment environment in utils

* Add `getDeploymentEnvironment()`: wrapper around `pulumi stack ls`
* Change `runPulumiCommand()` to syncronous

Commit:
Use new helpers in `infra:deploy`

Commit:
Add new `infra:environment:get` command

This command displays the pulumi stack currently selected
for this project. It's a wrapper around `pulumi stack ls` command

Commit:
Rename deployments to infra

And change functions' name accordingly. The idea behind the change
is because we are not storing only deployments in the file.

Commit:
Change getDeploymentEnvironment to getPulumiStack

Commit:
Add `deployed` and change variables name

Commit:
Add call to addInfra()

Commit:
Add removeInfra()

Commit:
Remove `utils` from dependencies

Avoid circular dependencies

Commit:
Add getEnvironmentDeployment()

Commit:
Add check for deployed environemts

Commit:
Change call to addInfra()

Commit:
Fix: check account sid

Commit:
Implement `infra:environment:set`

Commit:
Implement `infra:environment:new`

vernig commented:
Tested with: 

* Deploy
  * Run `twilio infra:deploy` 
  * Check that `.twilio-infra` is added 
* Attempt to deploy to the wrong environment
  * After deployment change twilio CLI profile to a different project. 
  * Run `twilio infra:deploy`
  * Should get an error saying that the environment is already associated to a different account
* Destroy 
  * Go back to the original Twilio project 
  * Run `twilio infra:destroy`
  * Check that `.twilio-infra` is removed
* Set a non existing environemnt
  * Run `twilio infra:environment:set fsdhjkhsdkjadh`
  * Should return `Environment fsdhjkhsdkjadh is not defined`
* Add a new environment to an existing deployment
  * Make sure the project selected in the CLI has an entry in `.twilio-infra`
  * Run `twilio infra:environment:new <name>`
  * Should get n error `Your Twilio project `ACxxxx` is already associated to another environment (xxx)`
* Add a new environment
  * Login to a new project in the CLI 
  * Run `twilio infra:environment:new prod`
  * Should succesfully create a new environment. Check that the eenvironment is added to `.twilio-infra` with `deployed = false`
 * Add a new environment to a non deployed project 
   * Run `twilio infra:environment:new prod2`
   * CLI should warn you that you are overwriting the environment associated to the project and prompt to continue

vernig commented:
@bermudezmt @kaiquelupo I started implementing the environments. The only test I performed so far are the one in the previous comment. Please feel free to add new comment(s) with other test cases you would like to implement.

Commit:
Bump-up version

Commit:
Add blank lines before and after output

Commit:
Interrupt execution is environment not defined

bermudezmt commented on src/commands/infra/environment/new.js:
```diff
@@ -0,0 +1,50 @@
+          addInfra(accountSid, getPulumiStack());
+        }
+        Printer.printSuccess(
+          'New environment created. When you are ready, use\n\n  $ twilio infra:deploy\n\nto deploy it to your Twilio prpject'
```
```suggestion
          'New environment created. When you are ready, use\n\n  $ twilio infra:deploy\n\nto deploy it to your Twilio project'
```

bermudezmt commented on src/commands/infra/destroy.js:
```diff
@@ -1,21 +1,16 @@
-        }
-      });
+      removeInfra(this.twilioClient.accountSid);
+      Printer.printSuccess('Resource(s) destroyed succesfully!');
```
```suggestion
      Printer.printSuccess('Resource(s) destroyed successfully!');
```

vernig replied:
Thanks @bermudezmt !

bermudezmt commented on src/commands/infra/deploy.js:
```diff
@@ -1,22 +1,29 @@
+    let deployment = getEnvironmentDeployment();
+    if (deployment && deployment !== this.twilioClient.accountSid) {
+      throw new TwilioCliError(
+        `The current stack is already deployed to ${deployment}. Please switch to that profile or define a new envirnoment`
```
```suggestion
        `The current stack is already deployed to ${deployment}. Please switch to that profile or define a new environment`
```

vernig replied:
Thanks @bermudezmt !

bermudezmt reviewed (changes requested):
Added some commit suggestions. 

Spellcheck, @vernig 😜

vernig committed:
Update src/commands/infra/environment/new.js

Co-authored-by: Maria Bermudez <bermudezmt@users.noreply.github.com>

vernig committed:
Update src/commands/infra/destroy.js

Co-authored-by: Maria Bermudez <bermudezmt@users.noreply.github.com>

vernig committed:
Update src/commands/infra/deploy.js

Co-authored-by: Maria Bermudez <bermudezmt@users.noreply.github.com>

Commit:
Merge branch 'feat-environment' of https://github.com/twilio-infra-as-code/plugin-twilio-infra into feat-environment

Commit:
Add check on creating a new environment

If the current Twilio project is already associated to an environment
but not deployed, we ask the developer to confirm if they want to
create a new environment

vernig commented:
@kaiquelupo @bermudezmt the last commit bd36e4b adds the check before creating a new environment. You can test this using the following step: 

* Add a new environment to a non deployed project
  * Create a new IaC project or use an existing one that has not been deployed
  * Run twilio infra:environment:new <new_environment_name> 
  * CLI should warn you that you are overwriting the environment associated to the project and prompt to continue

```
The current Twilio project (ACxxxxx) is already associated to environment "xxxx", but it's not deployed. 
Are you sure you want to associate a new environment to it? Y/n
````
I updated the [comment above](https://github.com/twilio-infra-as-code/plugin-twilio-infra/pull/17#issuecomment-823469048) with the new test.

Commit:
Fix: check existing deployment environment

Commit:
Add `destroyInfra()`

This method set a deployment status to false in .twilio-infra

Commit:
Use destroyInfra instead of removeInfra

Commit:
Fix typo

bermudezmt commented:
Works beautifully, thank you!
